src/wrapper/clap-juce-wrapper.cpp

```diff
@@ -212,7 +212,8 @@ class ClapJuceWrapper : public clap::helpers::Plugin<clap::helpers::Misbehaviour
                 if (isBeingDestroyed())
                     return;
 
-                _host.latencyChanged();
+                if (_host.canUseLatency())
```

Comment:
I guess `_host.canFoo()` probably needs to be called on the main thread, otherwise it would be nice to avoid queueing an action for the main thread if we can't do the action anyway.

Reply:
So the cans can in theory be called off the main thread but it sort of made me nervous to have a check which leads to an assert non-thread-aligned with the item. That said the plugin just checks a variable and its functions are set up in the candos.

I think it will be a rare case when these aren't implemented, so i did it this way. Also I wanted the code to read that every _host.foo has a _host.canfoo right next to it.

what do you think? I can go either way. Sort of 'edge case event suppression' vs 'clearer code when you copy and paste it' type thing?

Reply:
Yeah, after thinking about it again, I think the way you currently have it is best, both in terms of thread-safety and readability :).